### PR TITLE
Issue #867: Support writing with RangePartitioning

### PR DESCRIPTION
**Pull request by vishalkarve15**

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ## 0.31.1 - 2023-06-06
 
+* Issue #867: Support writing with RangePartitioning
```
Comment (davidrabinowitz) [inline CHANGES.md:7]: This line should be in the `Next` section

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java [hunk 2]
```diff
@@ -905,6 +906,31 @@ private void testPartition(String partitionType) {
...
+        .option("partitionRangeStart", "1")
+        .option("partitionRangeEnd", "21")
+        .option("partitionRangeInterval", "2")
+        .option("partitionRequireFilter", "true")
+        .option("table", table)
+        .option("writeMethod", writeMethod.toString())
+        .save();
+
+    Dataset<Row> readDF = spark.read().format("bigquery").load(table);
+    assertThat(readDF.count()).isEqualTo(3);
```
[the remark this answers is not in the document]
Comment (vishalkarve15) [reply spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/WriteIntegrationTestBase.java:933]: Done.

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java [hunk 3]
```diff
@@ -592,6 +593,12 @@ public void loadDataIntoTable(
       options.getPartitionField().ifPresent(timePartitionBuilder::setField);
       jobConfiguration.setTimePartitioning(timePartitionBuilder.build());
     }
+    if (options.getPartitionField().isPresent() && options.getPartitionRange().isPresent()) {
+      RangePartitioning.Builder rangePartitionBuilder = RangePartitioning.newBuilder();
+      rangePartitionBuilder.setField(options.getPartitionField().get());
```
Comment (davidrabinowitz) [inline bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:598]: nit:
```
options.getPartitionField().ifPresent(rangePartitionBuilder::setField);
options.getPartitionRange().ifPresent(rangePartitionBuilder::setRange);
```

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
Please update `README-template.md` with the new options, and explain the difference between date and range partitioning in the `Configuring Partitioning` section

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**